azurerm/internal/services/kusto/tests/kusto_cluster_resource_test.go

```diff
@@ -261,6 +284,46 @@ resource "azurerm_kusto_cluster" "test" {
 `, data.RandomInteger, data.Locations.Primary, data.RandomString)
 }
 
+func testAccAzureRMKustoCluster_keyVault(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+data "azurerm_client_config" "current" {}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_key_vault" "test" {
+  name                = "acctestkc%s-vault"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  tenant_id           = data.azurerm_client_config.current.tenant_id
+  sku_name            = "standard"
+}
+
+resource "azurerm_kusto_cluster" "test" {
+  name                = "acctestkc%s"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+
+  sku {
+    name     = "Dev(No SLA)_Standard_D11_v2"
+    capacity = 1
+  }
+
+  key_vault {
+    key_name      = "acctestkc%s"
+    key_version   = "1"
```

Comment:
Hi @tombuildsstuff 

> .. it sounds like the Kusto <-> Key Vault association is going to need to be pulled out into it's own resource, since the Key Vault Access Policy needs to be created after the Kusto Identity has been created. The `azurerm_storage_account_customer_managed_key` resource works in the same way for the same reason - as such would you be able to split this association out into a separate resource?

I also think that a separate resource is necessary. I'll change this PR to be able to support this feature.